neuronpp/core/cells/spine_cell.py

```diff
@@ -78,3 +120,369 @@ def _connect_necks_rand_uniform(neck: Sec, sections):
                 neck.hoc.connect(s(loc), 0.0)
                 added[s.name()].append(loc)
                 break
+
+
+    def add_spines_at(self, dist_range, spine_density,
+                      spine_type="generic", **kwargs):
```

Comment:
I'm not a great fan of `kwargs`, although it's sometimes useful, in the line 456 you named `kwargs` as `mechs_with_gbar_name` so maybe you can do the same here? so it would be `spine_params` rather than `kwargs`?